src/AddressSpace.cc

```diff
@@ -881,22 +884,33 @@ void AddressSpace::unmap_internal(remote_ptr<void> addr, ssize_t num_bytes) {
 
     Mapping m = move(mm);
     mem.erase(m.map);
+
+    // Also unmap any corresponding mapping from the local address space
+    if (m.local_addr) {
+      uint8_t* start_addr =
+          m.local_addr + (max(m.map.start(), rem.start()) - m.map.start());
+      uint8_t* stop_addr =
+          m.local_addr + (min(m.map.end(), rem.end()) - m.map.end());
+      munmap(start_addr, stop_addr - start_addr);
+    }
```

Comment:
I think it would be the caller's responsibility to update the local mapping in that case. Of course we'd have to give them a way to update the local mapping address(es). For now, clearing the local address(es) sounds fine to me.